services/stats/misc.go

```diff
@@ -0,0 +1,25 @@
+package stats
+
+import (
+	"strings"
+
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/utils/types/deployment"
+)
+
+// TODO: Do we need this?
+func CleanupTagsBasedOnDeploymentType(tags Tags, key string) Tags {
```

Comment:
@chandumlg I remember @cisse21 mentioning that we shall push all tags (unfiltered) and infra shall take care of removing tags where necessary. Do I remember correctly?

Reply:
Let's keep this like this for now. There technical reasons were filter on infra might not be feasible.

Reply:
I will keep it, but in another form, decoupling the stats package from this logic through a config option (`excludedTags`)